x-pack/platform/packages/shared/kbn-failure-store-modal/src/components/retention_period_field/retention_period_field.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FunctionComponent } from 'react';
+import React, { useState } from 'react';
+import { EuiButton, EuiPopover, EuiSelectable } from '@elastic/eui';
+
+import { UseField } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+
+import { NumericField } from '@kbn/es-ui-shared-plugin/static/forms/components';
+import { timeUnits, extraTimeUnits } from '../constants';
+
+export const RetentionPeriodField: FunctionComponent<{ disabled: boolean }> = ({ disabled }) => {
+  const [open, setOpen] = useState(false);
+  return (
+    <UseField
+      path="retentionPeriodValue"
+      component={NumericField}
+      componentProps={{
+        fullWidth: true,
+        euiFieldProps: {
+          disabled,
+          'data-test-subj': `selectFailureStorePeriodValue`,
+          min: 1,
+          append: (
+            <UseField key={'retentionPeriodUnit'} path={'retentionPeriodUnit'}>
+              {(field) => {
+                const onSelect = (option: string) => {
+                  field.setValue(option);
+                  setOpen(false);
+                };
+
+                return (
+                  <EuiPopover
+                    isOpen={open}
+                    panelPaddingSize="none"
+                    closePopover={() => setOpen(false)}
+                    button={
+                      <EuiButton
+                        data-test-subj="selectFailureStoreRetentionPeriodUnit"
+                        disabled={disabled}
+                        iconType="arrowDown"
+                        iconSide="right"
+                        color="text"
+                        onClick={() => setOpen((isOpen) => !isOpen)}
+                      >
+                        {[...timeUnits, ...extraTimeUnits].find(
```

Comment:
nit: feel free to ignore, I just thought might be clearer if we do something like


```suggestion
                        const allUnits = useMemo(() => [...timeUnits, ...extraTimeUnits], []);
                        const labelFor = useCallback(
                          (val: string) => allUnits.find((u) => u.value === val)?.text ?? val,
                          [allUnits]
                        );

                        ... 
                        {labelFor(field.value)}
```